### PR TITLE
feat: add subscriptionType

### PR DESCRIPTION
**Pull request by jasonkuhrt**
supports https://github.com/graphql-nexus/nexus/issues/447

### About the implementation

- graphql.js treats subscription type as an object type
- at least, for example, `GraphQLFieldConfig` type in graphql has optional `subscribe` field

	image

- for this reason, in builder, we share the `buildObjectType` in Nexus, and it just works

	Screenshot of playground for local sandbox app using linked `@nexus/schema` and working

	image

- I briefly considered reusing the existing `NexusObjectTypeDef` and genericizing its field config type –– but I found this to add more complexity and type errors across the codebase as I began in that direction.

	For example, the union of being either subscription config or object config made `implements` method need rethinking.

	image

### Note

I do not think the story about Nexus & subscription support is done with this PR.

For example the typings on the resolve method are `any` which is wrong––the `root` param on resolvers of subscription root fields should be typed to the return value of the `subscribe` field.

And that is already an existing issue with [`subscriptionField`](https://github.com/graphql-nexus/schema/issues/454).

I strongly believe we should iterate instead of letting this PR live for long.

### CODE REVIEW

File: src/definitions/subscriptionType.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   */
+  resolve(
+    root: T,
+    args: ArgsValue<TypeName, FieldName>,
+    context: GetGen<'context'>,
+    info: GraphQLResolveInfo
+  ):
+    | MaybePromise<ResultValue<'Subscription', FieldName>>
+    | MaybePromiseDeep<ResultValue<'Subscription', FieldName>>
+}
```
Comment (Weakky) [inline src/definitions/subscriptionType.ts:31]: I know it's not really part of the pr goal, but can we maybe make `resolve` optional and provide an "identity" default value?

Comment (Weakky) [reply src/definitions/subscriptionType.ts:31]: Saw your comment on the PR. Ok to iterate further down the road

File: src/builder.ts [hunk 2]
```diff
@@ -286,6 +289,7 @@ export type TypeToWalk =
   | { type: 'named'; value: GraphQLNamedType }
   | { type: 'input'; value: NexusShapedInput }
   | { type: 'object'; value: NexusShapedOutput }
+  | { type: 'subscription'; value: NexusSubscriptionTypeConfig }
```
Comment (Weakky) [inline src/builder.ts:292]: Given that subscription is just an object type, what's the reason for needing an additional type to walk here? It looks like it's not even walking the type in a special way. Can't we just type the surface API but then return a simple `ObjectTypeDef` ?

File: src/definitions/subscriptionType.ts (hunk 1 above)
Comment (jasonkuhrt) [reply src/definitions/subscriptionType.ts:31]: Funny you say that --> https://github.com/graphql-nexus/schema/issues/78#issuecomment-654005333

File: src/definitions/subscriptionType.ts [hunk 3]
```diff
@@ -74,9 +74,7 @@ export interface SubscribeFieldConfig<TypeName extends string, FieldName extends
 
 export interface SubscriptionDefinitionBuilder extends ObjectDefinitionBuilder<'Subscription'> {}
 
-export class SubscriptionDefinitionBlock<
-  TypeName extends string = 'Subscription'
-> extends ObjectDefinitionBlock<TypeName> {
+export class SubscriptionDefinitionBlock extends ObjectDefinitionBlock<'Subscription'> {
```
Comment (jasonkuhrt) [inline src/definitions/subscriptionType.ts:77]: @Weakky can you explain why this fix worked?

**Pull request merged**